src/components/Results.tsx

```diff
@@ -117,6 +124,29 @@ export const Results: React.FC<{
     return null;
   }
 
+  const searchResultsRefs = useMemo(
+    () =>
+      filteredFacilities.reduce((refs, facility) => {
+        const ref = createRef<HTMLDivElement>();
+        return { ...refs, [facility.id]: ref };
+      }, {}),
+    []
+  );
+
+  const handleFacilitySelection = useCallback((facilityId: string) => {
```

Comment:
In such a case `useCallback` is useless, you can simply define a regular callback